quint/src/ir/namespacer.ts

```diff
@@ -0,0 +1,83 @@
+/* ----------------------------------------------------------------------------------
+ * Copyright (c) Informal Systems 2023. All rights reserved.
+ * Licensed under the Apache 2.0.
+ * See License.txt in the project root for license information.
+ * --------------------------------------------------------------------------------- */
+
+/**
+ * Recursive addition of namespaces into all names in IR components.
+ *
+ * @author Gabriela Moreira
+ *
+ * @module
+ */
+
+import { IRTransformer, transformDefinition } from './IRTransformer'
+import { QuintApp, QuintDef, QuintLambda, QuintName, isFlat } from './quintIr'
+import { QuintConstType } from './quintTypes'
+
+/**
+ * Adds a namespace to a QuintDef and the names inside it, preserving a set of names.
+ *
+ * @param def - The QuintDef to add the namespace to.
+ * @param namespace - The namespace to add.
+ * @param namesToPreserve - A set of names that should not receive the namespace (i.e. builtin names).
```

Comment:
You say here builtin names, but you actually call it with `Flattener.currentModuleNames`

```suggestion
 * @param namesToPreserve - A set of names that should not receive the namespace (e.g. builtin names).
```